src/app/ui/course/course.component.ts

```diff
@@ -131,6 +131,12 @@ export class CourseComponent implements OnInit {
   updateFavoriteCourse() {
     this.favorite$.pipe(take(1)).toPromise().then(fav => this.auth.updateFavoriteCourse(this.id, !fav));
   }
+
+  editCourseDetails() {
+    this.db.updateCourseName(this.course.id, this.course.name);
```

Comment:
Please use NgxSpinnerService (look for other usages in the project) and show spinner during the update, hide it when it is done. Use Promise.all